plugin-api/src/main/java/org/sonar/api/web/AbstractUrlPattern.java

```diff
@@ -112,6 +112,10 @@ private static String convertPatternsToString(List<String> input) {
     Builder() {
     }
 
+    /**
+     * @deprecated since 10.0. Products implementing the API should define this internally.
+     */
+    @Deprecated(since = "10.0")
```

Comment:
Should we track on SQ/SC to not rely on this anymore?

Reply:
that's already the case in SQ. SC will just have to copy this when upgrading the plugin API. I'm not sure if it's worth creating a dedicated ticket for it.